app/javascript/menu/main-menu.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { SideNav } from 'carbon-components-react/es/components/UIShell';
+
+import { FirstLevel } from './first-level';
+import { GroupSwitcher } from './group-switcher';
+import { MenuCollapse } from './menu-collapse';
+import { MenuSearch } from './search';
+import { MiqLogo } from './miq-logo';
+import { SearchResults } from './search-results';
+import { SecondLevel } from './second-level';
+import { Username } from './username';
+
+
+const initialExpanded = window.localStorage.getItem('patternfly-navigation-primary') !== 'collapsed';
+
+export const MainMenu = (props) => {
+  const { applianceName, currentGroup, currentUser, customBrand, logoLarge, logoSmall, menu, miqGroups, showLogo, showUser } = props;
+  const [expanded, setExpanded] = useState(initialExpanded);
+  const [activeSection, setSection] = useState(null);
+  const [searchResults, setSearch] = useState(null);
+
+  const hideSecondary = (e) => setSection(null);
+
+  let appearExpanded = expanded || !!activeSection || !!searchResults;
+
+  useEffect(() => {
+    window.localStorage.setItem('patternfly-navigation-primary', expanded ? 'expanded' : 'collapsed');
+  }, [expanded]);
+
+  useEffect(() => {
+    const classNames = {
+      true: 'miq-main-menu-expanded',
+      false: 'miq-main-menu-collapsed',
+    };
+    document.body.classList.remove(classNames[!appearExpanded]);
+    document.body.classList.add(classNames[appearExpanded]);
+  }, [appearExpanded]);
+
+  useEffect(() => {
+    // cypress, debugging
+    ManageIQ.menu = menu;
+  }, [menu]);
+
+  return (
+    <>
+      <div
+        onClick={hideSecondary}
+      >
```

Comment:
I think if you have a single attribute, it's more readable if this is a one-liner.